lib/field_definition_table.erb

```diff
@@ -9,44 +9,39 @@
   </thead>
   <tbody>
     <% for item in data[:fields] %>
-
-      <!-- TODO: Externalize the styling in this erb so that it is consistent with the rest of the site. Keeping in
-                 mind that the table as a whole needs to not scroll. Individual examples should, but the entire table
-                 should not. It looks tacky -->
-      <tr>
-        <td>
-          <!-- Render the field name as a link, if a url is provided.-->
-          <% if !item[:url].nil? %>
-            <a target="_blank" href="<%= item[:url]%>"><code><%= item[:name]%></code></a>
-          <% else %>
-            <code><%= item[:name]%></code>
-          <% end %>
-        </td>
-        <td><%= item[:required] %> </td>
-        <td><%= item[:cardinality] %> </td>
-        <td><%= item[:type] %> </td>
-      </tr>
-
-      <tr>
-        <td colspan="4">
-        <ul style="list-style-type: none; margin: 1.2em">
-          <% unless item[:description].nil? %>
-              <li><i>Description</i></li>
-              <ul style="list-style-type: none"><li><%= item[:description] %></li></ul>
-          <%end%>
-
-          <% unless item[:example].nil? %>
-              <li><i>Example</i></li>
-              <ul style="list-style-type: none"><li><pre><%= item[:example] %></pre></li></ul>
-          <%end%>
-
-          <% unless item[:note].nil? %>
-              <li><i>Notes</i></li>
-              <ul style="list-style-type: none"><li><%= item[:note] %></li></ul>
-          <%end%>
-        </ul>
-        </td>
-      </tr>
+    <tr>
+      <td>
+        <a name="<%= "#{data[:table_name]}-#{item[:name]}" %>">
+        <% if !item[:url].nil? %>
+          <a target="_blank" href="<%= item[:url]%>"><code><%= item[:name]%></code></a>
+        <% else %>
+          <code><%= item[:name]%></code>
+        <% end %>
+        </a>
+      </td>
+      <td><%= item[:required] %> </td>
+      <td><%= item[:cardinality] %> </td>
+      <td><%= item[:type] %> </td>
+    </tr>
+    <tr>
+      <td colspan="4">
+      <ul style="list-style-type: none; margin: 1.2em">
+      <% unless item[:description].nil? %>
+        <li><i>Description</i></li>
+        <ul style="list-style-type: none"><li><%= item[:description] %></li></ul>
+      <%end%>
+      <% unless item[:example].nil? %>
+        <li><i>Example</i></li>
+        <ul style="list-style-type: none"><li><pre><%= item[:example] %></pre></li></ul>
+      <%end%>
+      <% unless item[:note].nil? %>
+        <li><i>Notes</i></li>
+        <ul style="list-style-type: none"><li><%= item[:note] %></li></ul>
+      <%end%>
+      </ul>
+      <div style="float: right"><a href="#<%=data[:table_name]%>">top</a></div>
```

Comment:
I can rename to 'top of section' but I do think it is necessary or at the least, desirable. I didn't add it until I was consistently getting annoyed when trying to actually use my own documentation and kept getting lost, especially after adding the Update section when suddenly I had 2 nearly identical tables and I wasn't sure which version of the table I was looking at.  

Ultimately I think it'd be better to use some scripting and render just one table with the ability to toggle between actions, but who has time for that? :)  And even then, with a sufficiently long table it would still be nice to get back up to the top quickly.

As for the placement of the text, since it isn't content it is intentionally out of the way as to be unobtrusive.